src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -0,0 +1,914 @@
+                    node.ArgumentRefKindsOpt,
+                    node.ArgsToParamsOpt,
+                    _localScopeDepth,
+                    _diagnostics);
+            }
+
+            return null;
+        }
+
+        public override BoundNode? VisitIndexerAccess(BoundIndexerAccess node)
+        {
+            using var _ = GetArgumentPlaceholders(node.ReceiverOpt, node.Arguments);
+            base.VisitIndexerAccess(node);
+
+            if (!node.HasErrors)
+            {
+                var indexer = node.Indexer;
+                CheckInvocationArgMixing(
+                    node.Syntax,
+                    indexer,
+                    node.ReceiverOpt,
+                    indexer.Parameters,
+                    node.Arguments,
+                    node.ArgumentRefKindsOpt,
+                    node.ArgsToParamsOpt,
+                    _localScopeDepth,
+                    _diagnostics);
+            }
+
+            return null;
+        }
+
+        public override BoundNode? VisitFunctionPointerInvocation(BoundFunctionPointerInvocation node)
+        {
+            using var _ = GetArgumentPlaceholders(receiverOpt: null, node.Arguments);
+            base.VisitFunctionPointerInvocation(node);
+
+            if (!node.HasErrors)
+            {
+                var method = node.FunctionPointer.Signature;
+                CheckInvocationArgMixing(
+                    node.Syntax,
+                    method,
+                    receiverOpt: null,
+                    method.Parameters,
+                    node.Arguments,
+                    node.ArgumentRefKindsOpt,
+                    argsToParamsOpt: default,
+                    _localScopeDepth,
+                    _diagnostics);
+            }
+
+            return null;
+        }
+
+        public override BoundNode? VisitAwaitExpression(BoundAwaitExpression node)
+        {
+            var placeholders = ArrayBuilder<(BoundValuePlaceholderBase, uint)>.GetInstance();
+            GetAwaitableInstancePlaceholders(placeholders, node.AwaitableInfo, GetValEscape(node.Expression, _localScopeDepth));
+            using var _ = new PlaceholderRegion(this, placeholders);
+            placeholders.Free();
+            base.VisitAwaitExpression(node);
+            return null;
+        }
+
+        private void GetAwaitableInstancePlaceholders(ArrayBuilder<(BoundValuePlaceholderBase, uint)> placeholders, BoundAwaitableInfo awaitableInfo, uint valEscapeScope)
+        {
+            if (awaitableInfo.AwaitableInstancePlaceholder is { } placeholder)
+            {
+                placeholders.Add((placeholder, valEscapeScope));
+            }
+        }
+
+        // Based on NullableWalker.VisitDeconstructionAssignmentOperator().
+        public override BoundNode? VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
+        {
+            base.VisitDeconstructionAssignmentOperator(node);
+
+            var left = node.Left;
```

Comment:
nit: can be inlined

Reply:
I'll leave this as is, for consistency with `right`.